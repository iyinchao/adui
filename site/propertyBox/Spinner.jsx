import React from "react"
import styles from "./components/propertyBox.scss"
import CodeBox from "./components/CodeBox"
import { Button, Switch, Spinner } from "../../components"
import { SIZES } from "./consts"

export default class PropertyBox extends React.Component {
  state = {
    hasLetter: false,
    size: "small",
  }

  render() {
    const { hasLetter, size } = this.state

    const codes = ` <Spinner ${hasLetter ? `text="加载中" ` : ""}${
      size === "small" ? "" : `size="${size}" `
    }/>`
    return (
      <div className={styles.wrapper}>
        <div className={styles.property}>
          <div className={styles.left}>
            <Spinner text={hasLetter ? "加载中" : null} size={size} />
          </div>
          <div className={styles.right}>
            <div className={styles.rightTitle}>配置项</div>
            <dl className={styles.controls}>
              <dt className={styles.controlTitle}>尺寸</dt>
              <dd className={styles.controlContent}>
                <Button.Group banner>
                  {SIZES.map(({ value, text }) => (
                    <Button
                      key={value}
                      active={value === size}
                      onClick={() => this.setState({ size: value })}
                    >
                      {text}
                    </Button>
                  ))}
                </Button.Group>
              </dd>
            </dl>
            <dl className={styles.controls}>
              <div className={styles.switchControl}>
                <Switch
                  checkedText="带文字提示"
                  unCheckedText="带文字提示"
                  checked={hasLetter}
                  onChange={(param) => this.setState({ hasLetter: param })}
                />
              </div>
            </dl>
          </div>
        </div>
        <CodeBox codes={codes} />
      </div>
    )
  }
}
