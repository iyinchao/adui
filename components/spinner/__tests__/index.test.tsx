import * as React from "react"
import { render } from "enzyme"
import TestRenderer from "react-test-renderer"
import sinon from "sinon"
import Spinner from ".."

describe("Spinner", () => {
  it("Spinner 是否正确渲染", () => {
    const wrapper = TestRenderer.create(
      <Spinner color="#00bb9c">children</Spinner>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it("value Prop 是否能验证 [0, 1] 外的数字", () => {
    const stub = sinon.stub(console, "error")
    render(<Spinner value={2} />)
    expect(stub.calledOnce).toEqual(true)
    stub.restore()
  })

  it("各情况快照核对", () => {
    const wrapper0 = TestRenderer.create(<Spinner size="mini" />)
    const wrapper1 = TestRenderer.create(<Spinner size="small" />)
    const wrapper2 = TestRenderer.create(<Spinner size="medium" />)
    const wrapper3 = TestRenderer.create(<Spinner size="large" />)
    const wrapper4 = TestRenderer.create(
      <Spinner value={0.27} text="指定进度" />
    )

    expect(wrapper0).toMatchSnapshot()
    expect(wrapper1).toMatchSnapshot()
    expect(wrapper2).toMatchSnapshot()
    expect(wrapper3).toMatchSnapshot()
    expect(wrapper4).toMatchSnapshot()
  })
})
