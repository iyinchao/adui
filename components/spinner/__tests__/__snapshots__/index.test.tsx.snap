// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Spinner 是否正确渲染 1`] = `
<div
  className="undefined adui-spinner-nestedWrapper"
>
  <div
    className="adui-spinner-base adui-spinner-normal adui-spinner-spinning"
    style={
      Object {
        "color": "#00bb9c",
      }
    }
  >
    <svg
      className="adui-spinner-circular"
      fill="none"
      height={20}
      viewBox="0 0 100 100"
      width={20}
    >
      <circle
        className="adui-spinner-track"
        cx="50%"
        cy="50%"
        r="44"
        strokeWidth={12.5}
      />
      <circle
        className="adui-spinner-head"
        cx="50%"
        cy="50%"
        r="44"
        strokeWidth={12.5}
        style={
          Object {
            "strokeDasharray": "282.743px",
          }
        }
      />
    </svg>
  </div>
  <div
    className="adui-spinner-container adui-spinner-blur"
    style={
      Object {
        "minHeight": 200,
      }
    }
  >
    children
  </div>
</div>
`;

exports[`Spinner 各情况快照核对 1`] = `
<div
  className="adui-spinner-base adui-spinner-normal adui-spinner-spinning"
  style={Object {}}
>
  <svg
    className="adui-spinner-circular"
    fill="none"
    height={16}
    viewBox="0 0 100 100"
    width={16}
  >
    <circle
      className="adui-spinner-track"
      cx="50%"
      cy="50%"
      r="44"
      strokeWidth={12.5}
    />
    <circle
      className="adui-spinner-head"
      cx="50%"
      cy="50%"
      r="44"
      strokeWidth={12.5}
      style={
        Object {
          "strokeDasharray": "282.743px",
        }
      }
    />
  </svg>
</div>
`;

exports[`Spinner 各情况快照核对 2`] = `
<div
  className="adui-spinner-base adui-spinner-normal adui-spinner-spinning"
  style={Object {}}
>
  <svg
    className="adui-spinner-circular"
    fill="none"
    height={20}
    viewBox="0 0 100 100"
    width={20}
  >
    <circle
      className="adui-spinner-track"
      cx="50%"
      cy="50%"
      r="44"
      strokeWidth={12.5}
    />
    <circle
      className="adui-spinner-head"
      cx="50%"
      cy="50%"
      r="44"
      strokeWidth={12.5}
      style={
        Object {
          "strokeDasharray": "282.743px",
        }
      }
    />
  </svg>
</div>
`;

exports[`Spinner 各情况快照核对 3`] = `
<div
  className="adui-spinner-base adui-spinner-normal adui-spinner-spinning"
  style={Object {}}
>
  <svg
    className="adui-spinner-circular"
    fill="none"
    height={28}
    viewBox="0 0 100 100"
    width={28}
  >
    <circle
      className="adui-spinner-track"
      cx="50%"
      cy="50%"
      r="44"
      strokeWidth={10.714285714285715}
    />
    <circle
      className="adui-spinner-head"
      cx="50%"
      cy="50%"
      r="44"
      strokeWidth={10.714285714285715}
      style={
        Object {
          "strokeDasharray": "282.743px",
        }
      }
    />
  </svg>
</div>
`;

exports[`Spinner 各情况快照核对 4`] = `
<div
  className="adui-spinner-base adui-spinner-normal adui-spinner-spinning"
  style={Object {}}
>
  <svg
    className="adui-spinner-circular"
    fill="none"
    height={38}
    viewBox="0 0 100 100"
    width={38}
  >
    <circle
      className="adui-spinner-track"
      cx="50%"
      cy="50%"
      r="44"
      strokeWidth={10.526315789473685}
    />
    <circle
      className="adui-spinner-head"
      cx="50%"
      cy="50%"
      r="44"
      strokeWidth={10.526315789473685}
      style={
        Object {
          "strokeDasharray": "282.743px",
        }
      }
    />
  </svg>
</div>
`;

exports[`Spinner 各情况快照核对 5`] = `
<div
  className="adui-spinner-base adui-spinner-normal"
  style={Object {}}
>
  <svg
    className="adui-spinner-circular"
    fill="none"
    height={20}
    viewBox="0 0 100 100"
    width={20}
  >
    <circle
      className="adui-spinner-track"
      cx="50%"
      cy="50%"
      r="44"
      strokeWidth={12.5}
    />
    <circle
      className="adui-spinner-head"
      cx="50%"
      cy="50%"
      r="44"
      strokeWidth={12.5}
      style={
        Object {
          "strokeDasharray": "282.743px",
          "strokeDashoffset": "206.40239px",
        }
      }
    />
  </svg>
  <div
    className="adui-spinner-text"
  >
    指定进度
  </div>
</div>
`;
